x-pack/auditbeat/module/system/package/package_homebrew.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build !windows
+
+package pkg
+
+import (
+	"bufio"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// InstallReceiptSource represents the "source" object in Homebrew's INSTALL_RECEIPT.json.
+type InstallReceiptSource struct {
+	Path string
+}
+
+// InstallReceipt represents the JSON object in Homebrew's INSTALL_RECEIPT.json.
+type InstallReceipt struct {
+	Source InstallReceiptSource
+}
+
+func listBrewPackages() ([]*Package, error) {
+	packageDirs, err := ioutil.ReadDir(homebrewCellarPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading directory %s", homebrewCellarPath)
+	}
+
+	var packages []*Package
+	for _, packageDir := range packageDirs {
+		if !packageDir.IsDir() {
+			continue
+		}
+		pkgPath := path.Join(homebrewCellarPath, packageDir.Name())
+		versions, err := ioutil.ReadDir(pkgPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error reading directory: %s", pkgPath)
+		}
+
+		for _, version := range versions {
+			if !version.IsDir() {
+				continue
+			}
+			pkg := &Package{
+				Name:        packageDir.Name(),
+				Version:     version.Name(),
+				InstallTime: version.ModTime(),
+			}
+
+			// Read formula
+			var formulaPath string
+			installReceiptPath := path.Join(homebrewCellarPath, pkg.Name, pkg.Version, "INSTALL_RECEIPT.json")
+			contents, err := ioutil.ReadFile(installReceiptPath)
+			if err != nil {
+				pkg.Error = errors.Wrapf(err, "error reading %v", installReceiptPath)
+			} else {
+				var installReceipt InstallReceipt
+				err = json.Unmarshal(contents, &installReceipt)
+				if err != nil {
+					pkg.Error = errors.Wrapf(err, "error unmarshalling JSON in %v", installReceiptPath)
+				} else {
+					formulaPath = installReceipt.Source.Path
+				}
+			}
+
+			if formulaPath == "" {
+				// Fallback to /usr/local/Cellar/{pkg.Name}/{pkg.Version}/.brew/{pkg.Name}.rb
+				formulaPath = path.Join(homebrewCellarPath, pkg.Name, pkg.Version, ".brew", pkg.Name+".rb")
+			}
+
+			file, err := os.Open(formulaPath)
```

Comment:
Is this handle closed?

Reply:
Thanks, I've added a `defer file.Close()`.